types.go

```diff
@@ -55,6 +55,11 @@ type (
 		// Checksum ensures the integrity of packages
 		// refereced by URL. Ignored for literals.
 		Checksum Checksum `json:"checksum"`
+	}
+
+	PackageRef struct {
+		PackageName      string
+		PackageNamespace string
```

Comment:
Namespace+Name uniquely identifies objects in the Kubernetes universe, equivalent to UID.  

It's worth noting that everything you see here is mutable. It's up to our implementation to cache correctly. However, immutability can be implemented as a matter of _policy_: VersionedFunction/VersionedPackage could avoid ever mutating a package.

I agree about the entrypoint.  I'll move it to FunctionSpec.

We want the Package mechanism to be more general than functions. Validation is a good point -- but I think we could do that at build time (or runtime, when there's no builder).